tests/installation/module_registration/register_extensions_and_modules.pm

```diff
@@ -11,12 +11,32 @@
 use base 'y2_installbase';
 use strict;
 use warnings;
-use testapi qw(save_screenshot get_required_var);
+use testapi qw(save_screenshot get_required_var get_var);
 
 sub run {
     my @scc_addons = grep($_, split(/,/, get_required_var('SCC_ADDONS')));
     $testapi::distri->get_module_registration()->register_extension_and_modules([@scc_addons]);
     save_screenshot;
+    my $timeout = 60 * get_var('TIMEOUT_SCALE', 1);
+    $testapi::distri->wait_registration_common_regcode_finished({timeout => $timeout, interval => 2, message => 'Page to insert module registration code did not appear'});
+    foreach my $addon (@scc_addons) {
```

Comment:
please, try to user high-order functions, not loops, specially not in testing layer. For example:
```
my @scc_addons = ('a', 'b', 'we');
my ($addon) = grep { /we|ha|ltss/ } @scc_addons;
print "SCC_REGCODE_" . uc $addon . "\n";
```

Reply:
we can unify those 3 methods `add_separate_*_registration_code` using a regex when we try to search it with libyui-rest-api at the level of Page.

Reply:
then we pass as parameter the particular addon and the controller method will check with a grep, for example if is 'we' so it would trigger at the end `get_module_regcode()->trust_gnupg_key()` because the other two methods are exactly the same. In that way in theory we could avoid to use any loop.

Reply:
> please, try to user high-order functions, not loops, specially not in testing layer. For example:
> 
> ```
> my @scc_addons = ('a', 'b', 'we');
> my ($addon) = grep { /we|ha|ltss/ } @scc_addons;
> print "SCC_REGCODE_" . uc $addon . "\n";
> ```

But if our case need both ha and ltss.  This method cannot handle this situation.